RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -185,8 +276,19 @@ private string CreateBaseWorkingDirectory()
             return baseDirectory;
         }
 
-        private IEnumerable<DiagnosticAnalyzer> GetAnalyzers(string packageRootDir, string additionalSearchFolder, string language)
+        private bool HasAnalyzers(IPackage package, string language)
+        {
+            return GetAnalyzers(package, language).Any();
```

Comment:
Once for the check that packages have analyzers, and once during the generation run. I'm not sure how one can combine them without keeping some kind of map <IPackage, IEnumerable<DiagnosticAnalyzer>>, which seems less-than-desirable.